tools/bindgen.sh

```diff
@@ -101,6 +101,18 @@ fc-bindgen amazonlinux-v5.10.y/arch/x86/include/asm/mpspec_def.h \
            >src/arch_gen/src/x86/mpspec.rs
 # https://github.com/rust-lang/rust-bindgen/issues/1274
 
+info "BINDGEN msr-index.h"
+cp -r amazonlinux-v5.10.y/include/asm-generic amazonlinux-v5.10.y/include/asm
```

Comment:
may I know where the sequence below comes from?

Reply:
Sure! That is required to fix include dependencies.
msr-index.h includes transitively `asm/rwonce.h`, but it actualy reside in [`asm-generic/rwonce.h`](https://github.com/torvalds/linux/blob/master/include/asm-generic/rwonce.h). This why it copies `asm-generic/` to `asm/`.

Let me explain other parts as well:
- `sed` command removes some attributes that `bindgen` does not know.
- `-Iamazonlinux-v5.10.y/include/` and `-Iamazonlinux-v5.10.y/arch/x86/include/` is requried to let `bindgen` know include paths.
- `-Wno-macro-redefined` is nice-to-have to suppress warning for `__always_inline` macro that does not affect bindings generation.
- `perl` command replaces decimal values with hex values. (`bindgen` seems to not have a option for that. If you have better solution, please let me know!)

Reply:
Thanks! I guess it is traceable here if someone needs to dig in later.